libs/platform/ebpf_object.c

```diff
@@ -326,3 +329,35 @@ ebpf_object_dereference_by_id(ebpf_id_t id, ebpf_object_type_t object_type)
     ebpf_lock_unlock(&_ebpf_object_tracking_list_lock, state);
     return return_value;
 }
+
+ebpf_result_t
+ebpf_object_reference_by_handle(
+    ebpf_handle_t handle, ebpf_object_type_t object_type, _Outptr_ ebpf_core_object_t** object)
```

Comment:
To me it seems confusing to have two APIs with almost identical names:
ebpf_object_reference_by_handle and ebpf_reference_object_by_handle.   That is, it seems too easy
to confuse one for the other.   It seems the difference is whether it passes in a type or not.  Suggest putting that in the function name, like `ebpf_object_reference_by_handle_and_type`

Reply:
Renamed one of them to `ebpf_reference_base_object_by_handle`.